Azure.ResourceManager.Core/AzureResourceManagerClient.cs

```diff
@@ -179,43 +166,12 @@ public T GetResourceOperations<T>(string subscription, string resourceGroup, str
                 CultureInfo.InvariantCulture) as T;
         }
 
-        /// <summary>
-        /// Gets resource operations base.
-        /// </summary>
-        /// <typeparam name="TContainer"> The type of the container class for a specific resource. </typeparam>
-        /// <typeparam name="TOperations"> The type of the operations class for a specific resource. </typeparam>
-        /// <typeparam name="TResource"> The type of the class containing properties for the underlying resource. </typeparam>
-        /// <param name="subscription"> The id of the Azure subscription. </param>
-        /// <param name="resourceGroup"> The resource group name. </param>
-        /// <param name="name"> The resource type name. </param>
-        /// <param name="model"> The resource data model. </param>
-        /// <param name="location"> The resource geo-location. </param>
-        /// <returns> Resource operations of the resource. </returns>
-        public ArmResponse<TOperations> CreateResource<TContainer, TOperations, TResource>(string subscription, string resourceGroup, string name, TResource model, LocationData location = default)
-            where TResource : TrackedResource
-            where TOperations : ResourceOperationsBase<TOperations>
-            where TContainer : ResourceContainerBase<TOperations, TResource>
-        {
-            if (location == null)
-            {
-                location = LocationData.Default;
-            }
-
-            TContainer container = Activator.CreateInstance(typeof(TContainer), ClientOptions, new GenericResourceData($"/subscriptions/{subscription}/resourceGroups/{resourceGroup}", location)) as TContainer;
-
-            return container.CreateOrUpdate(name, model);
-        }
-
-        /// <summary>
-        /// Gets default subscription.
-        /// </summary>
-        /// <param name="cancellationToken"> A token to allow the caller to cancel the call to the service. The default value is <see cref="P:System.Threading.CancellationToken.None" />. </param>
-        /// <returns> A <see cref="Task"/> that on completion returns the subscription id. </returns>
-        internal async Task<Subscription> GetDefaultSubscriptionAsync(CancellationToken cancellationToken = default)
+        private Subscription GetDefaultSubscription()
         {
-            return await GetSubscriptionContainer().GetDefaultSubscriptionAsync(cancellationToken);
+            var sub = GetSubscriptionContainer().List().FirstOrDefault();
+            if (sub is null)
```

Comment:
using "is" to avoid using "==" due to overload?

Reply:
yes